### PR TITLE
Add removeParent to EditTeam

### LINKED ISSUE #1267: Not possible to update a child team to be top-level
Due to the `omitempty` tag on `NewTeam.ParentTeamID` it is not possible (in my understanding) to update a child team to be a top-level team. Because updating a team is a PATCH operation, when `parent_team_id` is omitted, GitHub does not update the team's parent.

Happy to submit a PR.

### PR DESCRIPTION
**Pull request by gmlewis**
Fixes #1267.

I wrote this PR to communicate my thoughts about one way to fix #1267 since I'm not sure I clearly communicated this idea when reviewing #1356.

### CODE REVIEW

**Comment by gmlewis** [comment]
Let's go ahead and have this PR reviewed as a possible fix for #1267.

File: github/teams.go [hunk 1]
```diff
@@ -187,9 +188,32 @@ func (s *TeamsService) CreateTeam(ctx context.Context, org string, team NewTeam)
...
+			LDAPDN       *string  `json:"ldap_dn,omitempty"`
+		}{
+			Name:        team.Name,
+			Description: team.Description,
+			Maintainers: team.Maintainers,
+			RepoNames:   team.RepoNames,
+			Privacy:     team.Privacy,
+			LDAPDN:      team.LDAPDN,
+		}
+		req, err = s.client.NewRequest("PATCH", u, teamRemoveParent)
```
Comment (wesleimp) [inline github/teams.go]: I'm not sure if it is the place for this code snippet. Maybe you can extract it to another function.

Comment (gmlewis) [reply github/teams.go]: I've moved it.  PTAL.

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Pull request merged**